### DISCUSSION TITLE
How to get just updated form field with updated value?

### DISCUSSION THREAD

**Discussion by ArchanaSharma95** [react-hook-form/react-hook-form#9911] [Q&A]
I build a form using this package. Everything is working fine except for my one requirement. I need to save every form field changes on blur(in case of input fields) and on change (for switch checkbox etc). Now have added a "watch" function to watch every form change.

But my requirement is to get the updated field with its updated value. How is this possible? Because I am getting all the form fields inside the watch function?

**Comment by Phibee** [comment]
> I build a form using this package. Everything is working fine except for my one requirement. I need to save every form field changes on blur(in case of input fields) and on change (for switch checkbox etc). Now have added a "watch" function to watch every form change.
> 
> But my requirement is to get the updated field with its updated value. How is this possible? Because I am getting all the form fields inside the watch function?

You can check this one.
[How can I get all values of dirtyfield? #3275](https://github.com/react-hook-form/react-hook-form/discussions/3275)

**Comment by leapful** [comment]
@ArchanaSharma95 @Phibee 

You can use `<form onChange />` event handler to achieve the requirement.

Working demo: https://codesandbox.io/s/late-framework-8o4m74?file=/src/App.js
